lib/icinga/checkable-check.cpp

```diff
@@ -512,6 +512,9 @@ void Checkable::ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrig
 			SetSuppressedNotifications(suppressed_types_after);
 		}
 	}
+
+	if (reachabilityChanged)
+		OnReachabilityChanged(this, cr, children, origin);
```

Comment:
`OnReachabilityChanged` is always called when there are children for every single check result, regardless if there even was some state change at all.

This in itself is not introduced by this PR, but is a problem  for the rest of this PR as it will send a full state update for each child including the write-through to MySQL. So each check result for some important checkable with lots of dependents could cause huge amounts of pointless MySQL row updates.

Reply:
"could cause"? Does it or does it not? I mean: don’t we have checksums?

Reply:
If I remember correctly, we only use those in the delta computation for the full initial sync and not for runtime updates.

Reply:
At least for me that's expected behavior in Icinga DB and the reason for why we're even distinguishing in the code whether we send a runtime update. In this example, Icinga 2 can just perform a simple check while Icinga DB would first have to additionally read the checksum back from the database (or keep a cache of these checksums).

Reply:
A great idea!